api/payments.go

```diff
@@ -148,25 +148,22 @@ func (a *API) PaymentCreate(w http.ResponseWriter, r *http.Request) error {
 		return internalServerError("We failed to authorize the amount for this order: %v", err)
 	}
 	tr := models.NewTransaction(order)
-
-	order.PaymentProcessor = provider.Name()
 	processorID, err := charge(params.Amount, params.Currency)
 	tr.ProcessorID = processorID
 
 	if err != nil {
 		tr.FailureCode = strconv.FormatInt(http.StatusInternalServerError, 10)
 		tr.FailureDescription = err.Error()
-		tr.Status = "failed"
-	} else {
-		tr.Status = "pending"
```

Comment:
do we not need a callback from the provider? Or does the charge take the payment all the way through?